widgets/views/shareLink.php

```diff
@@ -1,32 +1,82 @@
+
+        window.open(url, 'share', opts);
+
+        return false;
+";
+
+$linkOptions = ['onclick' => $option, 'target' => '_blank', 'rel' => 'noopener noreferrer'];
+
+/** @var SocialShareService $socialShareService */
+$socialShareService = new SOcialShareService;
 ?>
-<div class="shareLinkContainer pull-right">
-    <?php
-    $option = "
-			var width = 575,
-				height = 400,
-				left = ($(window).width() - width) / 2,
-				top = ($(window).height() - height) / 2,
-				url = this.href;
-				opts = 'status=1' +
-	                ',width=' + width +
-	                ',height=' + height +
-	                ',top=' + top +
-	                ',left=' + left;
-
-	            window.open(url, 'share', opts);
-
-	            return false;
-
-			";
-    ?>
-    <?= Html::a('<i class="fa fa-facebook" style="font-size:16px;color:#3a5795">&nbsp;</i>', 'https://www.facebook.com/sharer/sharer.php?u=' . urlencode((string)$permalink) . '&description=' . urlencode((string)$object->getContentDescription()), ['onclick' => $option]); ?>
-    <?= Html::a('<i class="fa fa-twitter" style="font-size:16px;color:#55acee">&nbsp;</i>', 'https://twitter.com/intent/tweet?text=' . urlencode((string)$object->getContentDescription()) . '&url=' . urlencode((string)$permalink), ['onclick' => $option]); ?>
-    <?= Html::a('<i class="fa fa-linkedin-square" style="font-size:16px;color:#0177b5">&nbsp;</i>', 'https://www.linkedin.com/shareArticle?summary=&mini=true&source=&title=' . urlencode((string)$object->getContentDescription()) . '&url=' . urlencode((string)$permalink) . '&ro=false', ['onclick' => $option]); ?>
-    <?= Html::a('<i class="fa fa-share" style="font-size:16px;color:#00c300">&nbsp;</i>', 'https://social-plugins.line.me/lineit/share?' . '&text=' . urlencode((string)$object->getContentDescription()) . '&url=' . urlencode((string)$permalink), ['onclick' => $option]); ?>
-</div>
+
+<span class="shareLinkContainer">
+    <div class="pull-right">
+        <?php if ($settings->facebook_enabled): ?>
+            <?= $socialShareService->createShareLink(
+                SocialShareService::PLATFORM_FACEBOOK,
+                $permalink,
+                $object->getContentDescription(),
+                array_merge($linkOptions, ['onclick' => $option])
+            ); ?>
+        <?php endif; ?>
+
+        <?php if ($settings->twitter_enabled): ?>
+            <?= $socialShareService->createShareLink(
+                SocialShareService::PLATFORM_TWITTER,
+                $permalink,
+                $object->getContentDescription(),
+                array_merge($linkOptions, ['onclick' => $option])
+            ); ?>
+        <?php endif; ?>
+
+        <?php if ($settings->linkedin_enabled): ?>
+            <?= $socialShareService->createShareLink(
+                SocialShareService::PLATFORM_LINKEDIN,
+                $permalink,
+                $object->getContentDescription(),
+                array_merge($linkOptions, ['onclick' => $option])
```

Comment:
the code `array_merge($linkOptions, ['onclick' => $option])` can be replaced with `$linkOptions`.

Reply:
Done